src/precomputed_quantities.jl

```diff
@@ -15,10 +15,14 @@ end
 function precomputed_quantities!(Y, p, t, colidx)
     ᶜuₕ = Y.c.uₕ
     ᶠw = Y.f.w
-    (; ᶜuvw, ᶜK, ᶜts, ᶜp, params) = p
-    (; ᶜinterp) = p.operators
+    (; ᶠuvw, ᶜuvw, ᶜK, ᶜts, ᶜp, params) = p
+    (; ᶠwinterp, ᶜinterp) = p.operators
+    ᶜJ = Fields.local_geometry_field(axes(Y.c.ρ)).J
     C123 = Geometry.Covariant123Vector
-    @. ᶜuvw[colidx] = C123(ᶜuₕ[colidx]) + C123(ᶜinterp(ᶠw[colidx]))
+    @. ᶜuvw[colidx] = C123(ᶜuₕ[colidx]) + C123(ᶜinterp(ᶠw[colidx])) # u̅
+    @. ᶠuvw[colidx] =
+        C123(ᶠwinterp(Y.c.ρ[colidx] * ᶜJ[colidx], ᶜuₕ[colidx])) +
+        C123(Y.f.w[colidx]) # ũ
```

Comment:
Interpolation needs to be done in contravariant components:
```suggestion
    @. ᶠuvw[colidx] =
       ᶠwinterp(Y.c.ρ[colidx] * ᶜJ[colidx], Geometry.Contravariant123Vector(ᶜuₕ[colidx])) +
        Geometry.Contravariant123Vector(Y.f.w[colidx]) # ũ
```